argo_rollouts/routing-traffic-by-using-argo-rollouts-for-openshift-service-mesh.adoc

```diff
@@ -0,0 +1,43 @@
+[id="routing-traffic-by-using-argo-rollouts-for-openshift-service-mesh"]
+= Routing traffic by using Argo Rollouts for OpenShift Service Mesh
+:context: routing-traffic-by-using-argo-rollouts-for-openshift-service-mesh
+
+toc::[]
+
+Argo Rollouts in {gitops-title} support various traffic management mechanisms such as xref:../argo_rollouts/routing-traffic-by-using-argo-rollouts.adoc#routing-traffic-by-using-argo-rollouts[OpenShift Routes] and Istio-based OpenShift Service Mesh.
+
+The choice for selecting a traffic manager to be used with Argo Rollouts depends on the existing traffic management solution that you are using to deploy cluster workloads. For example, Red{nbsp}Hat OpenShift Routes provides basic traffic management functionality and does not require the use of a sidecar container. However, Red{nbsp}Hat OpenShift Service Mesh provides more advanced routing capabilities by using Istio but also requires the configuration of a sidecar container.
+
+You can use OpenShift Service Mesh to split traffic between two application versions:
+
+* *Canary version*: A new version of an application where you gradually route the traffic.
+* *Stable version*: The current version of an application. After the canary version is stable and has all the user traffic directed to it, it becomes the new stable version. The previous stable version is discarded.
+
+The Istio-support within Argo Rollouts uses the Gateway and VirtualService resources to handle traffic routing.
+
+* *Gateway*: You can use a Gateway to manage inbound and outbound traffic for your mesh. The gateway is the entry point of OpenShift Service Mesh and handles traffic requests sent to an application.
+
+* *VirtualService*: VirtualService defines traffic routing rules and the percentage of traffic that goes to underlying services, such as the stable and canary services. 
+
+For example, in a sample deployment scenario, in the initial instance 100% of the traffic is directed towards the stable version of the application. The application is running as expected, and no additional attempts are made to deploy a new version.
+
+image::446_OpenShift_Service_Mesh_Argo_Rollouts_0724_1.png[100% of traffic in the stable version of the application]
+
+However, after deploying a new version of the application, Argo Rollouts creates a new canary deployment based on the new version of the application and routes some percentage of traffic to that new version.
+
+When using Service Mesh, Argo Rollouts automatically modifies the VirtualService resource to control the traffic split percentage between the stable and canary application versions. In the following diagram, 20% of traffic is sent to the canary application version after the first promotion and then 80% is sent to the stable version by the stable service.
+
+image::446_OpenShift_Service_Mesh_Argo_Rollouts_0724_2.png[80% of traffic in the stable version and 20% in the canary version]
+
+[id="prerequisites_{context}"]
```

Comment:
Why is this section here and not in the Procedure? Needs to be properly modularized.